skimage/util/tests/test_bouding_box_crop.py

```diff
@@ -0,0 +1,53 @@
+import numpy as np
+import pytest
+
+from skimage import data
+from skimage.util import bounding_box_crop
+
+rng = np.random.default_rng()
+
+
+def test_2d_crop_1():
+    data = rng.random((50, 50))
+    out_data = bounding_box_crop(data, [(0, 25)])
+    np.testing.assert_array_equal(out_data, data[:25, :])
```

Comment:
Hello, looking at these unitary tests, this PR doesn't seem to address the feature request #5428 :confused: !
Looking at the description of the feature request of @sofroniewn, the `bounding_box` parameter should be of the form `[(i0, j0), (i1, j1)]` and the function must return `image[i0: i1, j0: j1]`, but this PR seems to implement the `bounding_box` parameter as `[(i0, i1), (j0, j1)]`.
I understand the original feature request, but I don't see the advantage of this implementation over simple array slicing...
Thank you BTW @immortal3, and sorry for my super-late review...

Reply:
`axes` argument is key selling point of this function as we can have many dimensions but if we only want to crop certain axes.

#5428, As you already mentioned, It is a simple re-arrangement of a tuple. i prefer `[(i0, i1), (j0, j1)]`, because all information (min, max boundary) related to the particular axis is in a single tuple inside the list.

Reply:
Usually bounding boxes are defined using
- the top-left corner and the down-right corner -> `(i0, j0), (i1, j1)`
- the top-left corner and height and width -> `(i0, j0), (h, w)`

Looking at the unitary tests, I don't see much of value compared to simple slicing:
```python
crop = data[:25, :]
# VS
crop = bounding_box_crop(data, [(0, 25)])
```

Reply:
FWIW I think this (the existing PR) is a good API, corresponding closely with NumPy, while ultimately giving the same functionality.